GitUI/GitUICommands.cs

```diff
@@ -1725,8 +1725,22 @@ private bool RunFileHistoryCommand(IReadOnlyList<string> args, bool showBlame)
                 filterByRevision = true;
             }
 
-            ShowModelessForm(owner: null, requiresValidWorkingDir: true, preEvent: null, postEvent: null,
+            // Similar to StartFileHistoryDialog(), tests works better with this setup
+            if (AppSettings.UseBrowseForFileHistory.Value)
+            {
+                ShowModelessForm(owner: null, requiresValidWorkingDir: true, preEvent: null, postEvent: null,
+                                () => new FormBrowse(commands: this, new BrowseArguments
+                                {
+                                    RevFilter = filterByRevision ? revision?.ObjectId.ToString() : null,
+                                    PathFilter = fileHistoryFileName,
+                                    SelectedId = revision?.ObjectId
+                                }));
+            }
+            else
+            {
+                ShowModelessForm(owner: null, requiresValidWorkingDir: true, preEvent: null, postEvent: null,
                 () => new FormFileHistory(commands: this, fileHistoryFileName, revision, filterByRevision, showBlame));
+            }
```

Comment:
```suggestion
            else
            {
                ShowModelessForm(owner: null, requiresValidWorkingDir: true, preEvent: null, postEvent: null,
                                 () => new FormFileHistory(commands: this, fileHistoryFileName, revision, filterByRevision, showBlame));
            }
```

Reply:
Can we get StyleCop to handle this?

Reply:
I'm not good enough? :)
I'm not aware of any rules for this situation.